clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveProducerCallback.java

```diff
@@ -0,0 +1,59 @@
+package com.linkedin.davinci.kafka.consumer;
+
+import com.linkedin.venice.kafka.protocol.KafkaMessageEnvelope;
+import com.linkedin.venice.kafka.protocol.Put;
+import com.linkedin.venice.message.KafkaKey;
+import com.linkedin.venice.writer.VeniceWriter;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+
+public class ActiveActiveProducerCallback extends LeaderProducerCallback {
+  private static final Runnable NO_OP = () -> {};
+  private Runnable onCompletionFunction = NO_OP;
+
+  public ActiveActiveProducerCallback(
+      LeaderFollowerStoreIngestionTask ingestionTask,
+      ConsumerRecord<KafkaKey, KafkaMessageEnvelope> sourceConsumerRecord,
+      PartitionConsumptionState partitionConsumptionState,
+      LeaderProducedRecordContext leaderProducedRecordContext,
+      int subPartition,
+      String kafkaUrl,
+      long beforeProcessingRecordTimestamp) {
+    super(
+        ingestionTask,
+        sourceConsumerRecord,
+        partitionConsumptionState,
+        leaderProducedRecordContext,
+        subPartition,
+        kafkaUrl,
+        beforeProcessingRecordTimestamp);
+  }
+
+  @Override
+  public void onCompletion(RecordMetadata recordMetadata, Exception exception) {
+    this.onCompletionFunction.run();
```

Comment:
Not a problem, more of a thought, is this right to be invoked prior to the super method?  I'm not so sure, it seems the super onCompletion in LeaderProducerCallback today does some sanity checks.  Not sure if it makes more sense to run those first or not.

Reply:
This is the ordering that used to happen at: https://github.com/linkedin/venice/pull/142/files#diff-9ee5c502e0f36bca0cb6efa87e4bd7a0ad5c027bbe8c8613185e0a7b2512582aL1296

I am maintaining the same ordering in the refactored implementation. IDK if there is any issue to this ordering... I don't see any, but maybe there is.